modin/core/dataframe/pandas/partitioning/axis_partition.py

```diff
@@ -377,6 +377,8 @@ def deploy_splitting_func(
             A list of pandas DataFrames.
         """
         dataframe = pandas.concat(list(partitions), axis=axis, copy=False)
+        # to reduce peak memory consumption
```

Comment:
How much does this reduce peak memory consumption?

Reply:
It depends on the size of the objects in `partitions`. Here we don't keep one extra copy of the entire dataframe until the end of the function.

Reply:
Do you have any results on a benchmark we run regularly? It would be great to look at.

Reply:
No, only for synthetic bench (roughly repeats what happens when performing an operation on axial partitions):
```
import numpy as np
import pandas as pd
import time

df1 = pd.DataFrame(np.random.rand(10**6, 2 * 10**2))  # ~ 1GB
df2 = pd.DataFrame(np.random.rand(10**6, 2 * 10**2))  # ~ 1GB
partitions = (df1, df2)

dataframe = pd.concat(list(partitions), axis=0, copy=False)
del partitions, df1, df2

result = dataframe.abs()

del dataframe

time.sleep(5)

# case with del save around ~4GB of peak memory consumption

```

I guess we can just look at the results of the built-in (in CI) memory consumption check?

Reply:
In timedf the memory consumption is calculated by computing peak memory consumption by checking the stats similar to htop by computing the memory in child thread for every 0.001 seconds.

The peak memory calculated is [printed](https://github.com/intel-ai/timedf/blob/master/timedf/report/run_params.py#L82) as `max_system_memory` at the end of benchmark run in timedf if we need to compare memory consumption on any of our regular benchmarks.